backstory/src/main/java/com/google/sps/servlets/StoryManager.java

```diff
@@ -0,0 +1,111 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.servlets;
+
+import com.google.api.client.http.ByteArrayContent;
+import com.google.api.client.http.GenericUrl;
+import com.google.api.client.http.HttpRequest;
+import com.google.api.client.http.HttpResponse;
+import com.google.api.client.http.HttpTransport;
+import com.google.api.client.http.javanet.NetHttpTransport;
+import com.google.auth.http.HttpCredentialsAdapter;
+import com.google.auth.oauth2.GoogleCredentials;
+import com.google.auth.oauth2.IdTokenCredentials;
+import com.google.auth.oauth2.IdTokenProvider;
+import java.io.IOException;
+import java.util.Scanner;
+import org.json.JSONObject;
+
+public final class StoryManager {
+  /**
+   * Makes a post request with a JSON including GPT2 Parameters
+   */
+  private static HttpResponse makePostRequestGPT2(
+      String serviceUrl, String prompt, int textLength, Double temperature) throws IOException {
+    // Obtain Credentials
+    GoogleCredentials credentials = GoogleCredentials.getApplicationDefault();
+
+    // Validate Credentials
+    if (!(credentials instanceof IdTokenProvider)) {
+      throw new IllegalArgumentException("Credentials are not an instance of IdTokenProvider.");
+    }
+
+    // Generate Authentication Token
+    IdTokenCredentials tokenCredential = IdTokenCredentials.newBuilder()
+                                             .setIdTokenProvider((IdTokenProvider) credentials)
+                                             .setTargetAudience(serviceUrl)
+                                             .build();
+
+    // Configure URL
+    GenericUrl genericUrl = new GenericUrl(serviceUrl);
+
+    // Form Adapter with Authentication token
+    HttpCredentialsAdapter adapter = new HttpCredentialsAdapter(tokenCredential);
+    HttpTransport transport = new NetHttpTransport();
+
+    // Form JSON body using generation parameters
+    String requestBody = "{\"length\": " + textLength
+        + ",\"truncate\": \"<|endoftext|>\", \"prefix\": \"" + prompt
+        + "\", \"temperature\": " + temperature + "}";
```

Comment:
Can a HashMap + gson be used here to make this easier to read and reduce risk of introducing formatting errors?

Reply:
Hashmap+gson won't work due to it not being string,string, will develop better implementation beyond MVP. Created method to do this outside of the other body for now.